### PR TITLE
Change the lingsting. No more monkeyz.

### PR DESCRIPTION
**Pull request by ghost**
Fixes issue #2809

### CODE REVIEW

**Comment by ghost** [comment]
This works, but it's a little hacky. Why don't we remove the options to turn then into non-human species instead of throwing an exception? ~~or we could do both if some hacker fucker breaks it~~

**Comment by ghost** [comment]
Well, that would require a much larger refactoring of how it works. Currently, the sting transforms the victim into whatever the selected genome is for the ling transforming. I'm not sure that the intent is to prevent lings from becoming monkies via this method, and completely changing it would prevent lings from absorbing monkeys. 

I'm willing to do it like that, but I didn't want to remove functionality that wasn't addressed in the issue.

**Comment by ghost** [comment]
Alright, then some ~~hacker fucker~~ can make a pull request if they want to. It's not like we're getting paid to make good code, seeing as the Byond engine already gives us a -50 handicap to anything we do.

**Comment by ghost** [comment]
Well, you're obviously welcome to suggest alternatives, but good programming practice is not to remove functionality that isn't problematic. It's not hacky, it's the right way of doing it in a large codebase.

I work for a large software engineering firm, and I see this to be completely valid. If you would like to criticize my solution, presenting a viable alternative is the professional way to go about it.

A change like you're suggesting would involve adding a separate genome selection for stinging as opposed to transforming, and then copying and changing the selection code to create a list that does not include nonhuman genomes. 50+ lines of code vs 5. This isn't hacky, it's smart, and the other way is wasteful.

**Comment by ghost** [comment]
I have my doubts that this is true because it seems to me that as we stare down the barrel of a shotgun of monolithic code, one of the last things we want to do is get lazy in maintaining it, lest we get lazy in understanding how it works and then have 900 issues that nobody knows how to fix like in /tg/station.

Building a code made out of hacks (and this really is a hack, seeing as it covers up a problem without removing it outright) is a short-term solution, and this is a very long-term project. I think the proper thing to do is to get input on refactoring the code to purge the problem outright, to encourage better standards of coding and to make an overall more well-designed game.

But I still suggest adding in the exception as a failsafe if some shit goes down IC.

I also think it's unprofessional to bring up your well-earned credentials in an amateur project, seeing as it's barely relevant to the discussion. I think your actions should speak more than your diploma, no matter how much work you put into earning it.

**Comment by ghost** [comment]
Again, I'd agree, but this is by no far a hacky solution. The only other method of doing it would add much more code overhead, and actually be bad code practice. Right now, the selection code is reusable. Reusability and case-dependent functionality with repeat code being shared is called polymorphism, and is one of the standing tenets of object oriented programming. If anything, this is GOOD programming practice rather than hacky practice.

**Comment by ghost** [comment]
I will admit I do not know enough about this to talk about it smartly and I pray somebody else does.

**Comment by ghost** [comment]
I definitely think good peer-review is important in large, monolithic projects, and I welcome VIABLE feedback on any PR I submit. Calling it "hacky" isn't helpful, but if you or anyone wants to give me specific ways they think it can be done better, then we definitely owe it to the playerbase to look at better alternatives.

**Comment by blobbernaut** [comment]
thanks

**Comment by Pyko1** [comment]
Wait, does this prevent you from transforming people to liggers/cats/moth/IPC/skeletons as well?

**Comment by ArcLumin** [comment]
liggers/cats/moth/IPC/skeletons aren't human children though. They're siblings

**Comment by GenericGoose** [comment]
i dont even know anymore

**Comment by Pyko1** [comment]
So yeah, not only does it stop monkey sting, it also prevents any non-human, aka ligger/ipc/moth/cat dna from being transmitted, which is pretty bad to me.

On an unreleated sidenote anyone transformed to cat/cluwne by on round events should still be allowed to defend themself act like a normal player.

**Comment by ArcLumin** [comment]
just make it so that doing a dna sting will automatically remove the monkey disability in the DNA.

More specifically, make the dna sting also humanize when it's stung. 

. = owner.humanize(TR_KEEPITEMS | TR_KEEPIMPLANTS | TR_KEEPORGANS | TR_KEEPDAMAGE | TR_KEEPVIRUS | TR_KEEPSE)

**Comment by ghost** [comment]
then couldn't you ling sting monkeys into humans?

**Comment by ArcLumin** [comment]
Yup. Which you can already do now

**Comment by ghost** [comment]
Fair enough

**Comment by ghost** [comment]
So, the consensus is that non-human stings are okay, just not monkeys? I'd seen someone complaining in the issue I referenced about catbeast stings. I'll change it to whatever is wanted, just went based on my knowledge of what the issue is.

**Comment by ghost** [comment]
Have adjusted to apply the humanize() function to the target if target is being transformed into a monkey. Does that satisfy the problem?

File: code/game/gamemodes/changeling/powers/tiny_prick.dm [hunk 1]
```diff
@@ -118,6 +117,8 @@
 		spawn(10)
 			user.real_name = NewDNA.real_name
 			NewDNA.transfer_identity(C, transfer_SE=1)
+			if(!istype(selected_dna.dna.species, /datum/species/human))
+				C.humanize(TR_KEEPITEMS | TR_KEEPIMPLANTS | TR_KEEPORGANS | TR_KEEPDAMAGE | TR_KEEPVIRUS | TR_KEEPSE) //To keep lingstings from adding the monkey disability.
```
Comment (ghost) [inline code/game/gamemodes/changeling/powers/tiny_prick.dm:121]: TR_KEEPSE would prevent lings from stinging people and giving them disabilities or does that not happen due to the if statement?

**Comment by TDcoolguy300** [comment]
Code looks fine

**Comment by Pyko1** [comment]
Imho humanising them is bad.

Lets assume you are a ligger ling, and you want to frame someone.
You can only turn them in to humans with your name, not liggers that look like you.

**Comment by ArcLumin** [comment]
Humanizing doesn't make them into the human mob, but a humanoid mob. This is the same code as monkey to humanoid.

**Comment by ghost** [comment]
Just to make it clear, I support the removal of lings being able to monkeyify people but keeping their ability to apply other disabilities and to make people into catbeasts. Provided that's the case I'll +1 the PR.

**Comment by Kokojos** [comment]
^
Just to make it clear, I support the removal of lings being able to monkeyify people but keeping their ability to apply other disabilities and to make people into catbeasts. Provided that's the case I'll +1 the PR.

Also, make sure humanize dosen't make everything drop.

**Comment by Pyko1** [comment]
in that case the changelog needs fixing as it says.

- tweak: "Changed changeling stings to no longer be able to transform players into monkeys or other non-humans."

**Comment by ghost** [comment]
@Pyko1 It should still work like that, just keep them from having the monkey disability.

@McBawbaggings The humanize call is applied after changing the SE, so the SE from the lingsting is kept.

File: code/game/gamemodes/changeling/powers/tiny_prick.dm (hunk 1 above)
Comment (ghost) [reply code/game/gamemodes/changeling/powers/tiny_prick.dm:121]: Doesn't happen because the humanize() call comes after the transfer_identity() call.

**Comment by ghost** [comment]
@Kokojos i approve of being turned into a furry

**Comment by ArcLumin** [comment]
Aren't you already one

**Comment by ghost** [comment]
SHOTS FIRED

**Comment by tabsiforu** [comment]
@ArcLumin What did you just say, you shitlord? You should know I received my degree in Art History summa cum laude and have been involved in numerous drum circles in public protests on wall street, and I have experienced over 300 instances of tasering. I am trained in graffiti and I'm the top megaphone user in the entire tumblr contingent. You are little but another capitalist phony. I will reduce you to tears with the finest slam poetry, the intensity of which you have never heard on God's green earth, I swear. You truly believe you can utter such ignorance on my livejournal? Reconsider, fool. As we liveblog I am gchatting my furry allies across the SJW combine and your reddit username is being passed around so you should ready yourself for the downvotes, homophobe. The downvotes that will cripple the paltry number that is your karma. You're finished, youth. I can blog anywhere, anytime, and I can make you cry in over seven hundred emotions, and that's just with my expansive array of emoticons and ascii, and that's just with my keyboard. Not only am I classically educated in freeform poetry, I also have the ability to call upon the full force and ferociousness of the tumblr activist army, and I will use it to it's full potential to eradicate your online presence off my dashboard, you shitlord. Would that you might have known what divine punishment your snarky little PM was about to unleash upon you, you might have logged off and thought better. But you were compelled, and you did so, and now your reddit reputation is going to hurt, you idiotic moron. I will rain hormones all around you and you will drown in your own man tears. You're gonna wish you never signed up to tumblr, shitlord.

**Comment by ghost** [comment]
Changelog updated. Ready for merging whenever you guys want.

**Comment by ArcLumin** [comment]
>tumblr warfare copy pasta
Yup, confirms it

**Comment by tabsiforu** [comment]
@ArcLumin Stop fursecuting me'me

**Comment by Pyko1** [comment]
I thought we were going to remove the meme copypasters from github.

**Comment by tabsiforu** [comment]
@Pyko1 you'd have to ban half the contributors

**Comment by Pyko1** [comment]
implying any of the contributors including me, have ever contributed anything really. All we do is piss on the code, when we aren't too busy removing features.

**Comment by ghost** [comment]
@Pyko1 I promise to try harder and make more contributions (once i nuke my master branch because i accidentally committed to it and that ruins every other branch)

**Pull request merged**